contents/newsletter/choosing-technologies.md

```diff
@@ -0,0 +1,115 @@
+
+This is what we've learned about choosing them all.
+
+## Start with a hair-on-fire problem burning problem
+
+Because of the many costs of adopting a technology, you need a good reason to do so; nice to have is not good enough. For us, this usually is one of three things:
+
+1. **Costs.** For example, our Infrastructure team projecting our AWS costs to rise well above standard benchmarks triggered us to look into solutions for our highest cost services. 
+2. **Infrastructure.** For example, reaching scaling limits for our current infrastructure. When adding "more boxes" won't fix the issue, its time to look for another solution.. 
+3. **Customers.** For example, customers demand a new feature other companies have like data exports.
+
+> **Related:** [How we decide what to build](https://newsletter.posthog.com/p/how-we-decide-what-to-build)
+
+## Find a fire-extinguishing solution to that problem
+
+The goal of technology is to solve that hair-on-fire problem. We find potential technologies that do this in one of two ways: 
+
+1. **What are other people using to solve the problem?** Technologies are by definition tools or services other people have used to solve problems. For example, part of the reason [we chose ClickHouse](/blog/how-we-turned-clickhouse-into-our-eventmansion) was that Cloudflare used it.
+
+2. **How have YOU solved this problem in the past?** There are a lot of details that are only revealed by implementing and using a technology. This means a lot of pitfalls you can avoid if you've already fallen in them once.
+
+Realistically, the evaluation process happens at the same time as figuring out the options. This is because some options can be written off immediately as they don't fit our requirements. For example, when we were looking for [scalable replacements for Postgres](/blog/how-we-turned-clickhouse-into-our-eventmansion) in our early days, we knew we needed it to be open source and self-hostable. This immediately cut down our list. 
+
+## Evaluate solutions autonomously
+
+The big advantage of PostHog is that we bias for action. [Small teams](https://newsletter.posthog.com/p/the-magic-of-small-engineering-teams) have the autonomy to go choose and test technologies. We don't have an elaborate planning and procurement process.
+
+Instead of filling out paper work, our team evaluates technologies as close to the real world as possible. This means means [testing in production (safely)](/product-engineers/testing-in-production), mimicking the real world, and building a proof of concept. 
+
+An example of this is how [Michael used real slow queries](https://www.youtube.com/watch?v=LrmK0vAG1ek) to test SQL parsing speed for C++ versus existing Python. The clear impact of the changes meant he could be confident adopting C++ as an option.
+
+![Speedy C++](https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2024_10_28_at_15_56_51_2x_68f0cca985.png)
```

Comment:
fwiw, I found this graph hard to parse. maybe vertical lines would be better to showing the before and after?